### DISCUSSION TITLE
Ability to add Custom tools

### DISCUSSION THREAD

**Discussion by bent-verbiage** [langflow-ai/langflow#68] [Ideas]
First of all, this is awesome work, thank you! Even if you'll never get to the full functionality that langchain & pure python will give you, it is still a great way to introduce langchain and its concepts to people, and I'm definitely going to be using this repo for demonstration purposes.

While I'm sure your backlog contains a lot of features, personally I think the most useful next feature would be 
1) the ability to add custom tools, or better documentation so that I can do export-to-json > add tool > import-json.
2) the AI/Human/System prompts that go with the ChatModels (gpt3.5 and 4).

**Comment by ogabrielluiz** [comment]
Thanks for the kind words! 

Those are some great insights. 

Number 1 was already one of our highest priorities but the documentation might help us while we develop the functionality for custom tools. 

The documentation would still require the functionality to be developed because you wouldn't be able to run the agent without and the Tool wouldn't show up in the app. We'll focus on that in our next steps.

Number 2 is something we haven't considered yet but we will put it on our to-do list. 

Please let us know if there's anything else you want us to look into.

**Comment by bent-verbiage** [comment]
Hey @ogabrielluiz, awesome work on the latest release(s)! 
With so much additional functionality, it would now be really good to have a bit more documentation 😄 . I'm really intrigued by the TimeTravelGuide chain for example 🤔

**Reply by ogabrielluiz** [reply]
Hi! 

The documentation will be our focus very soon. We're just setting up the project to ease the process of outside contribution. 

We've refactored almost every system and we are very close to building a good developer experience. 

The docs should me similar to what you can do with LangChain just a bit limited(not for too long).

Anyways, we'll make some videos and create a documentation page aimed at developers and users.

**Comment by ippolite** [comment]
Hello, @ogabrielluiz! It seems that you've deployed exactly the same tool I was thinking about two weeks ago! I also have some custom tool implementation ideas as well as extra features. What would be the best way to have a chat with you?
